File: src/main/java/no/nav/syfo/metric/Metric.java
package no.nav.syfo.metric;

import io.micrometer.core.instrument.MeterRegistry;
import io.micrometer.core.instrument.Tags;
import org.springframework.stereotype.Controller;

import javax.inject.Inject;

@Controller
public class Metric {

    private final MeterRegistry registry;

    @Inject
    public Metric(MeterRegistry registry) {
        this.registry = registry;
    }

    public void countEvent(String name) {
        registry.counter(
                addPrefix(name),
                Tags.of("type", "info")
        ).increment();
    }

    public void tellHttpKall(int kode) {
        registry.counter(
                addPrefix("httpstatus"),
                Tags.of(
                        "type", "info",
                        "kode", String.valueOf(kode)
                )
        ).increment();
    }

    private String addPrefix(String navn) {
        String METRIKK_PREFIX = "syfotilgangskontroll_";
        return METRIKK_PREFIX + navn;
    }
}

File: src/main/kotlin/no/nav/syfo/metric/Metric.kt
package no.nav.syfo.metric

import io.micrometer.core.instrument.MeterRegistry
import io.micrometer.core.instrument.Tags
import org.springframework.stereotype.Component
import javax.inject.Inject

@Component
class Metric @Inject constructor(
    private val registry: MeterRegistry,
) {
    fun countEvent(name: String) {
        registry.counter(
            addPrefix(name),
            Tags.of("type", "info")
        ).increment()
    }

    fun tellHttpKall(kode: Int) {
        registry.counter(
            addPrefix("httpstatus"),
            Tags.of(
                "type",
                "info",
                "kode",
                kode.toString()
            )
        ).increment()
    }

    private fun addPrefix(navn: String): String {
        val metricPrefix = "syfotilgangskontroll_"
        return metricPrefix + navn
    }
}
